### PR TITLE
Ensure MapCache disposes cleanly.

### LINKED ISSUE #7248: Cross-mod texture bogosity
After switching from TD to RA, I found the RA menu background was rendered using the minimap sheet texture instead of the expected artwork.

I'm not sure how to repro this, but I suspect it is a regression from #6830.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7430430-cross-mod-texture-bogosity?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by RoosterDragon**
This prevents the map loading thread writing to disposed objects which can have unintended side effects.

Fixes #7248 - I have included more details and some reproduction steps against that issue if you need them.

### CODE REVIEW

**Comment by obrakmann** [comment]
I'm not able to reproduce this, but the code changes make sense. :+1:

**Comment by pchote** [comment]
Confirmed that this fixes #7248, and code looks fine :+1:.

File: OpenRA.Game/Map/MapCache.cs [hunk 1]
```diff
@@ -250,7 +250,20 @@ IEnumerator IEnumerable.GetEnumerator()
...
+			if (previewLoaderThread == null)
+			{
+				sheetBuilder.Dispose();
+				return;
+			}
+
+			// We need to let the loader thread exit before we can dispose our sheet builder.
+			// Ideally we should dispose our resources before returning, but we don't to block waiting on the loader thread to exit.
+			// Instead, we'll queue disposal to be run once it has exited.
+			ThreadPool.QueueUserWorkItem(_ =>
```
Comment (pchote) [inline OpenRA.Game/Map/MapCache.cs]: All our other code that needs a worker will explicitly spawn a new thread.  Do we want to convert those to use ThreadPool in a future PR?

**Comment by reaperrr** [comment]
Using this branch I couldn't reproduce #7781 anymore (most likely the same issue as #7248 anyway), so :+1: from me, too.

File: OpenRA.Game/Map/MapCache.cs (hunk 1 above)
Comment (RoosterDragon) [inline OpenRA.Game/Map/MapCache.cs]: That or Tasks would probably be a good idea, yes.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/c8f79032352efcab72ddd859076ead87382d176a%5E...c8f79032352efcab72ddd859076ead87382d176a)

**Pull request merged**